### DISCUSSION TITLE
magick can convert jpg -> heic, but can not convert heic -> jpg

### DISCUSSION THREAD

**Discussion by icesttail** [ImageMagick/ImageMagick#3808] [Help]
magick -version
Version: ImageMagick 7.0.10-59 Q16 x86_64 2021-01-16 ### https://imagemagick.org
Copyright: (C) 1999-2021 ImageMagick Studio LLC
License: https://imagemagick.org/script/license.php
Features: Cipher DPC HDRI OpenMP(4.5) 
Delegates (built-in): heic jbig jng jpeg ltdl lzma png tiff webp x zlib

magick -list format | grep HEIC
     HEIC* HEIC      rw+   High Efficiency Image Format (1.7.0)
     HEIF* HEIC      rw+   High Efficiency Image Format (1.7.0)

magick convert test.heic test1.jpg
convert: Unsupported feature: Unsupported codec `test.heic' @ error/heic.c/IsHeifSuccess/135.

Can anyone tell me what I'm missing.

**Comment by fmw42** [comment]
With ImageMagick 7, use magick, not magick convert.  Try

`magick test.heic test1.jpg`

**Reply by icesttail** [reply]
The same result,

magick test.heic test1.jpg
magick: Unsupported feature: Unsupported codec `test.heic' @ error/heic.c/IsHeifSuccess/135.

**Reply by fmw42** [reply]
Did you create the HEIC from a JPG or get it from a camera? If the former, try creating it again with 

```
magick test.jpg test.heic
magick test.heic test2.jpg

```

If this does not work, then try upgrading ImageMagick. Perhaps your version was prior to current development of HEIC and might not have supported writing at your version.  Do you have the most current version of libheif?

**Reply by icesttail** [reply]
Thank you for your advice. I updated libheif(1.7.0->1.12.0), and now it works.

**Comment by fmw42** [comment]
The ImageMagick web site lists HEIC as Read only. But I am not sure if this is outdated or not.


"HEIC | R | Apple High efficiency Image Format | Set the quality to 100 to produce lossless HEIC images. Requires the&nbsp;libheif&nbsp;delegate library."

See https://imagemagick.org/script/formats.php

**Reply by fmw42** [reply]
Libheif is now at v1.12. So your version is old. Perhaps upgrade it and try again.
